File: Examples/VBasic/GroupDocs.Text.Examples.VBasic/Utilities/WordStatistic.vb
Imports GroupDocs.Text.Extractors.Text

Public Class WordStatistic

    Public Sub New(fileName As String, maxWordLength As Integer)
        'ExStart:WordStatistic
        Dim factory As New ExtractorFactory()
        Dim statistic As New Dictionary(Of String, Integer)()

        Dim extractor As TextExtractor = factory.CreateTextExtractor(fileName)
        If extractor Is Nothing Then
            Console.WriteLine("The document's format is not supported")
            Return
        End If

        Try
            Dim line As String = Nothing
            Do
                line = extractor.ExtractLine()
                If line IsNot Nothing Then
                    Dim words As String() = line.Split(" "c, ","c, ";"c, "."c)
                    For Each w As String In words
                        Dim word As String = w.Trim().ToLower()
                        If word.Length > maxWordLength Then
                            If Not statistic.ContainsKey(word) Then
                                statistic(word) = 0
                            End If

                            statistic(word) += 1
                        End If
                    Next
                End If
            Loop While line IsNot Nothing
        Finally
            extractor.Dispose()
        End Try

        Console.WriteLine("Top words:")

        For i As Integer = 0 To 9
            Dim count As Integer = -1
            Dim maxKey As String = Nothing
            For Each key As String In statistic.Keys
                If statistic(key) > count Then
                    count = statistic(key)
                    maxKey = key
                End If
            Next

            If maxKey Is Nothing Then
                Exit For
            End If

            Console.WriteLine("{0}: {1}", maxKey, count)
            statistic.Remove(maxKey)
            'ExEnd:WordStatistic
        Next
    End Sub
    Public Shared Sub FindMaxWordLength(fileOne As String, fileTwo As String)
        'ExStart:FindMaxWordLength
        Dim firstFile As [String] = Common.getFilePath(fileOne)
        Dim secondFile As [String] = Common.getFilePath(fileTwo)
        Dim arguments As String() = New String() {firstFile, secondFile}

        Dim maxWordLength As Integer
        For i As Integer = 0 To arguments.Length - 1
            If arguments(i).Length = 1 OrElse Not Integer.TryParse(arguments(i), maxWordLength) Then
                maxWordLength = 5
            End If
            Dim ws As New WordStatistic(arguments(i), maxWordLength)
            Console.WriteLine("__________________")
        Next
        'ExEnd:FindMaxWordLength
    End Sub

End Class

File: Examples/CSharp/GroupDocs.Text.Examples.CSharp/Utilities/WordStatistic.cs
using GroupDocs.Text;
using GroupDocs.Text.Extractors.Text;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Text_for_.NET
{
    class WordStatistic
    {
        public WordStatistic(string fileName, int maxWordLength)
        {
            //ExStart:WordStatistic
            ExtractorFactory factory = new ExtractorFactory();
            Dictionary<string, int> statistic = new Dictionary<string, int>();

            TextExtractor extractor = factory.CreateTextExtractor(fileName);
            if (extractor == null)
            {
                Console.WriteLine("The document's format is not supported");
                return;
            }

            try
            {
                string line = null;
                do
                {
                    line = extractor.ExtractLine();
                    if (line != null)
                    {
                        string[] words = line.Split(' ', ',', ';', '.');
                        foreach (string w in words)
                        {
                            string word = w.Trim().ToLower();
                            if (word.Length > maxWordLength)
                            {
                                if (!statistic.ContainsKey(word))
                                {
                                    statistic[word] = 0;
                                }

                                statistic[word]++;
                            }
                        }
                    }
                }
                while (line != null);
            }
            finally
            {
                extractor.Dispose();
            }

            Console.WriteLine("Top words:");

            for (int i = 0; i < 10; i++)
            {
                int count = -1;
                string maxKey = null;
                foreach (string key in statistic.Keys)
                {
                    if (statistic[key] > count)
                    {
                        count = statistic[key];
                        maxKey = key;
                    }
                }

                if (maxKey == null)
                {
                    break;
                }

                Console.WriteLine("{0}: {1}", maxKey, count);
                statistic.Remove(maxKey);
            }
            //ExEnd:WordStatistic
        }
        public static void FindMaxWordLength(string fileOne, string fileTwo)
        {
            //ExStart:FindMaxWordLength
            String firstFile = Common.GetFilePath(fileOne);
            String secondFile = Common.GetFilePath(fileTwo);
            string[] arguments = new string[] { firstFile, secondFile};
           
            int maxWordLength;
            for (int i = 0; i < arguments.Length; i++)
            {
                if (arguments[i].Length == 1 || !int.TryParse(arguments[i], out maxWordLength))
                {
                    maxWordLength = 5;
                }
                WordStatistic ws = new WordStatistic(arguments[i], maxWordLength);
                Console.WriteLine("__________________");
            }
            //ExEnd:FindMaxWordLength
        }
    }
}
